### PR TITLE
further canonicalize names used in haproxy config

### PR DESCRIPTION
**Pull request by macb**
also adds the first tests :checkered_flag:

### CODE REVIEW

File: config/config_test.go [hunk 1]
```diff
@@ -1 +1,476 @@
...
+	"k8s.io/kubernetes/pkg/client/unversioned/testclient"
+	"k8s.io/kubernetes/pkg/util/intstr"
+)
+
+func testDir(t *testing.T) (string, func()) {
+	dir, err := ioutil.TempDir("", "haproxy_config")
+	if err != nil {
+		t.Fatal("failed to build dir")
+	}
+	return dir, func() { defer os.RemoveAll(dir) }
```
Comment (fatih) [inline config/config_test.go]: Shouldn't this be just `func() { os.RemoveAll(dir) }` if you're going to call it in a defer below ?

File: config/config_test.go [hunk 2]
```diff
@@ -1 +1,476 @@
...
+}
+
+func TestNewConfig(t *testing.T) {
+	dir, cleanup := testDir(t)
+	defer cleanup()
+
+	c := NewConfig(&testclient.FakeIngress{}, dir+"/file", "local.cluster")
+	if c == nil {
```
Comment (fatih) [inline config/config_test.go]: I think `NewConfig` should return an error if there is a possibility of it returning a nil pointer.

File: config/config_test.go (hunk 1 above)
Comment (macb) [inline config/config_test.go]: why yes, yes it should!

**Comment by tam7t** [comment]
lgtm :+1:

**Comment by tam7t** [comment]
still :+1:

**Pull request merged**